src/App.tsx

```diff
@@ -1,26 +1,244 @@
+  };
+
+  const clearCompleted = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    Promise.all(completedTodos.map(todo => deleteTodo(todo.id)));
+  };
+
+  const hideError = () => {
+    setErrorMessage('');
+  };
+
+  const hasCompleted = todos.some(todo => todo.completed);
+  const notCompletedCount = todos.filter(todo => !todo.completed).length;
+
+  const filteredTodos = todos.filter(todo => {
+    if (filter === 'active') {
+      return !todo.completed;
+    }
+
+    if (filter === 'completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <form onSubmit={addTodo}>
+            <input
+              type="text"
+              ref={inputRef}
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={title}
+              onChange={e => setTitle(e.target.value)}
+              disabled={isLoading}
+              data-cy="NewTodoField"
+            />
+          </form>
+        </header>
+
+        {todos.length > 0 && (
+          <TodoList
+            todos={filteredTodos}
+            tempTodo={tempTodo}
+            processingIds={processingIds}
+            onDelete={deleteTodo}
+          />
+        )}
+
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {notCompletedCount} items left
+            </span>
+            <div className="filters" data-cy="Filter">
+              <a
+                href="#/"
+                className={`filter__link ${filter === 'all' ? 'selected' : ''}`}
+                data-cy="FilterLinkAll"
+                onClick={() => setFilter('all')}
+              >
+                All
+              </a>
+              <a
+                href="#/active"
+                className={`filter__link ${filter === 'active' ? 'selected' : ''}`}
+                data-cy="FilterLinkActive"
+                onClick={() => setFilter('active')}
+              >
+                Active
+              </a>
+              <a
+                href="#/completed"
+                className={`filter__link ${filter === 'completed' ? 'selected' : ''}`}
+                data-cy="FilterLinkCompleted"
+                onClick={() => setFilter('completed')}
+              >
+                Completed
+              </a>
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method